src/runtime.js

```diff
@@ -209,6 +209,21 @@ function adoptShared(global, callback) {
   // locking).
   for (let i = 0; i < preregisteredExtensions.length; i++) {
     const fnOrStruct = preregisteredExtensions[i];
+    if (IS_ESM) {
+      // If we're in a module runtime, trying to execute a nomodule extension
+      // simply remove the nomodule extension so that it is not executed.
+      if (!fnOrStruct.e) {
```

Comment:
I think if we handled this check in `maybeLoadCorrectVersion`, we could squeeze a few more bytes out of it.

Reply:
done. moved to `maybeLoadCorrectVersion`